p2p/README.md

```diff
@@ -71,7 +71,22 @@ See [`raintree/router.go`](./raintree/router.go) for the specific implementation
 
 ## Module Architecture
 
+### Architecture Design Language
```

Comment:
Please add a TODO to move this (and the definitions you have below) into `shared` since this text is not just relevant to `p2p`